File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() => 40;
    public int RemainingMinutesInOven(int current) => ExpectedMinutesInOven() - current;
    public int PreparationTimeInMinutes(int layers) => layers * 2;
    public int ElapsedTimeInMinutes(int layers, int minutesBaked) => PreparationTimeInMinutes(layers) + minutesBaked;
}

File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven: int = 40

let remainingMinutesInOven (ellapsedTime: int): int  =
    expectedMinutesInOven - ellapsedTime

let preparationTimeInMinutes (numberLayers: int) : int = 
    numberLayers * 2

let elapsedTimeInMinutes (numberLayers: int) ( ellapsedTime: int): int =
    preparationTimeInMinutes numberLayers + ellapsedTime
